### PR TITLE
Adloox RTD Module: add new RTD module

### PR DESCRIPTION
**Pull request by jimdigriz**
## Type of change

- [X] New [RTD provider](https://github.com/corememltd/Prebid.js/blob/feature/adloox/modules/adlooxRtdProvider.js) ([documentation](https://github.com/corememltd/Prebid.js/blob/feature/adloox/modules/adlooxRtdProvider.md))

## Description of change

Introduction of Adloox's RTD provider module.

* contact email of the adapter’s maintainer: contact@adloox.com and alex+adloox@coremem.com
* this is an official submission

Disclosures:

 * loads the generic segmentation script (~3kB gzipped from `p.adlooxtracking.com`) and provides no user orientated segments (only page, slot quality and IVT/fraud related); used to provide a consistent experience with our [GPT](https://developers.google.com/publisher-tag/guides/get-started) clients as it also includes observability/tracing for our monitoring purposes which would not be appropriate to be included here.

This module automate the complex tasks an AdOps teams must perform when integrating with Adloox, including the script loading listed in the disclosure above. If the any of the scripts fail to load, it impacts only Adloox's measurement and does not affect the operation of Prebid.js or inventory fill for the publisher.

## Other information

 * Looking for feedback on what to do about `getGlobal().adUnits` in the RTD module; I looked to use `auctionManager.getAdUnits()` but it does not seem to be initialised early on enough to have any adunits in there and though [supposedly a no-no](https://docs.prebid.org/dev-docs/module-rules.html#global-module-rules) I see the other RTD modules are using `getGlobal()` so I just copied them on this one. Is there something better I could be doing here?
 * requires and communicates with the Adloox analytics module over a command processing queue
   * this module will not function without it and must not be merged before https://github.com/prebid/Prebid.js/pull/6308 has been merged
 * [raised but not elaborated on was the setting of the ad server key-value targetting](https://github.com/prebid/Prebid.js/pull/6174#discussion_r557606362) so left as is until someone chirps up

This PR was created on request in https://github.com/prebid/Prebid.js/pull/6174#issuecomment-763964406

Changes since that version:
 * word smithing
 * removal of `waitForIt` check
 * reduce 700ms auction delay recommendation to 100ms
 * [change eslint "warn" to "off"](https://github.com/prebid/Prebid.js/pull/6174#issuecomment-758582001) to neuter 'prebid/validate-imports' grumblings over the analytics command queue

### CODE REVIEW

**Comment by aleksatr** [comment]
@jimdigriz , circleci output
[adloox.txt](https://github.com/prebid/Prebid.js/files/6023368/adloox.txt)

**Comment by jimdigriz** [comment]
> @jimdigriz , circleci output [adloox.txt](https://github.com/prebid/Prebid.js/files/6023368/adloox.txt)

Confusing as what I get running the test locally is below, can you tell me what I am doing wrong?

```
alex@aineko:~/src/adloox/Prebid.js$ git log -n 1
commit 04c12f33ca1b11200a6b8b7db7f413e6fe6d31bc (HEAD -> feature/adloox, origin/feature/adloox-rtd, origin/feature/adloox, coremem/feature/adloox-rtd, coremem/feature/adloox, feature/adloox-rtd)
Author: Alexander Clouter <alex@coremem.com>
Date:   Sat Feb 13 16:37:26 2021 +0000

    Adloox real time data module

alex@aineko:~/src/adloox/Prebid.js$ rm -rf node_modules
alex@aineko:~/src/adloox/Prebid.js$ npm install
alex@aineko:~/src/adloox/Prebid.js$ gulp test
[16:40:30] Using gulpfile ~/src/adloox/Prebid.js/gulpfile.js
[16:40:30] Starting 'test'...
[16:40:30] Starting 'clean'...
[16:40:30] Finished 'clean' after 53 ms
[16:40:30] Starting 'lint'...
[16:40:52] Finished 'lint' after 22 s
[16:40:52] Starting 'test'...

START:
⚠ ｢wdm｣: 
ℹ ｢wdm｣: Compiled with warnings.
ℹ ｢wdm｣: Compiling...
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
⚠ ｢wdm｣: 
ℹ ｢wdm｣: Compiled with warnings.
22 02 2021 16:41:30.570:INFO [karma-server]: Karma v4.4.1 server started at http://0.0.0.0:9876/
22 02 2021 16:41:30.571:INFO [launcher]: Launching browsers ChromeHeadless with concurrency 5
22 02 2021 16:41:30.715:INFO [launcher]: Starting browser ChromeHeadless
22 02 2021 16:41:31.450:INFO [HeadlessChrome 88.0.4324 (Linux 0.0.0)]: Connected on socket a6wZ-FcbZ9GbzjhTAAAA with id 44231630

Finished in 8.633 secs / 5.222 secs @ 16:41:41 GMT+0000 (Greenwich Mean Time)

SUMMARY:
✔ 7587 tests completed
ℹ 5 tests skipped
[16:41:41] Finished 'test' after 49 s
[16:41:41] Finished 'test' after 1.18 min
```

**Comment by jimdigriz** [comment]
Assuming this PR requires work to align it to the work going on in https://github.com/prebid/Prebid.js/pull/6293 (Fpd 2.0 Update) and move to using `ortb2Imp` instead?

Once that work is done and merged, it will provide me something to rebase against so I would prefer to wait till then.

**Comment by jimdigriz** [comment]
Anything I need to do here to get this merged?

Slightly concerned as it has now been a month (in this PR along) and there have been no comments about the code its-self or what may need changing. If something is needed here it would be great to hear about it sooner rather than later.

Let me know if I can help at all. Thanks

**Comment by gglas** [comment]
waiting for 6308 to resolve, dependencies between modules

**Comment by jimdigriz** [comment]
@aleksatr any chance of reviewing commit https://github.com/prebid/Prebid.js/pull/6310/commits/65e0a9bb3717e77509c6b039d65298e3e2962495 for inclusion which helps towards unlocking this PR for merging; from our perspective as the authors.

**Comment by aleksatr** [comment]
hey @jimdigriz , changes to `src/targeting.js` and `src/ajax.js` should be in a separate PR, changes to the PBJS core have a bit different requirements, like reviews from two members.
On the other topic, I still see adlooxAdServer changes here, I assume that this PR will be rebased once 6309 get's merged?

**Comment by jimdigriz** [comment]
> hey @jimdigriz , changes to `src/targeting.js` and `src/ajax.js` should be in a separate PR, changes to the PBJS core have a bit different requirements, like reviews from two members.
> On the other topic, I still see adlooxAdServer changes here, I assume that this PR will be rebased once 6309 get's merged?

As requested these are in https://github.com/prebid/Prebid.js/pull/7160 and https://github.com/prebid/Prebid.js/pull/7161

**Review by aleksatr — APPROVED** [review]
@jimdigriz I'll merge after 6309 is merged and this one rebased

**Comment by jimdigriz** [comment]
@aleksatr anything I can do to help here?

**Comment by jimdigriz** [comment]
@aleksatr @jsnellbaker anything I need to do to get this reviewed and finally merged?

**Comment by ChrisHuie** [comment]
@jimdigriz I will make sure this gets merged in today.

**Review by ChrisHuie — APPROVED** [review]

**Pull request merged**
